### PR TITLE
CVE-2016-5130 and CVE-2016-1679

### PR DESCRIPTION
**Pull request by BenjaminHunt**
CVE-2016-5130 complete
CVE-2016-1679 complete

### CODE REVIEW

File: cves/CVE-2016-5130.yml [hunk 1]
```diff
@@ -60,7 +74,9 @@ fixes:
   :note: ''
 - :commit: 4d78d33cb1852e7d487f1d327517c4c4044a3ca3
   :note: ''
-vccs: []
+vccs:
```
Comment (meyersbs) [inline cves/CVE-2016-5130.yml]: The VCC(s) are very important for curating this data -- try and do a little digging and pinpoint when the Blink dependency was introduced.

File: cves/CVE-2016-5130.yml [hunk 2]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: Improper Access Control
```
Comment (meyersbs) [inline cves/CVE-2016-5130.yml]: Please include the CWE number here as well.

Comment (BenjaminHunt) [reply cves/CVE-2016-5130.yml]: Done.

File: cves/CVE-2016-5130.yml (hunk 1 above)
Comment (BenjaminHunt) [reply cves/CVE-2016-5130.yml]: I've briefly looked into it a little bit, but it is difficult to find. There are 15,200 code results, and 209,967 commit results when I search "Blink." The VCE points to the file that addresses the problem, not the file that uses it or calls the dependency, so I don't know how to find it.

If I think of a way to find where the dependency was introduced I will do it. If you have any suggestions, please let me know.

File: cves/CVE-2016-1679.yml [hunk 3]
```diff
@@ -32,7 +32,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  The ToV8Value function has unneccessary calls of setter functions that may
+  crash the Google Chrome Web Browser. This is do to a lack of necessary
```
Comment (Josh-Akers) [inline cves/CVE-2016-1679.yml]: Incorrect use of the word "do", should be "due".

File: cves/CVE-2016-1679.yml [hunk 4]
```diff
@@ -95,20 +100,23 @@ discovered:
...
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    A class used V8ValueConverter to convert a DictionaryValue to a v8::Value
+    with an untrustworth ScriptContext. This allows attackers to use a setter
```
Comment (Josh-Akers) [inline cves/CVE-2016-1679.yml]: Spelling error, add a 'y' to the end of "untrustworth".

File: cves/CVE-2016-1679.yml [hunk 5]
```diff
@@ -32,7 +32,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
```
Comment (Josh-Akers) [inline cves/CVE-2016-1679.yml]: Description feels like it's missing something.  What does this vulnerability allow and why is it a problem?

The initial bug report here should be of some help.  

https://bugs.chromium.org/p/chromium/issues/detail?id=606390

File: cves/CVE-2016-1679.yml [hunk 6]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: Other (NVD-CWE-Other)
```
Comment (Josh-Akers) [inline cves/CVE-2016-1679.yml]: I was under the impression that the CWE was supposed to be a number categorized by cwe.mitre.org.

Just go through the list for something that's related to memory not being freed.  

http://cwe.mitre.org/data/definitions/699.html

File: cves/CVE-2016-5130.yml (hunk 1 above)
Comment (Josh-Akers) [reply cves/CVE-2016-5130.yml]: Not sure if this helps but in 09/2015 the Blink repository was merged into Chromium.

File: cves/CVE-2016-5130.yml [hunk 7]
```diff
@@ -99,10 +111,11 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
+  answer: |
+    A chromium developer discovered the flaw in the
+  date: 2013-04-30
```
Comment (Josh-Akers) [inline cves/CVE-2016-5130.yml]: Are you sure this is accurate?  That's three years before the bug was reported.  Also, if this bug is being caused by the added dependency of Blink to chromium, then this bug wouldn't have existed prior to the introduction of those new dependencies would it?

File: cves/CVE-2016-5130.yml [hunk 8]
```diff
@@ -60,7 +70,9 @@ fixes:
   :note: ''
 - :commit: 4d78d33cb1852e7d487f1d327517c4c4044a3ca3
   :note: ''
-vccs: []
+vccs:
+- :commit:
+  :note: 'Whenever the Blink dependency was introduced'
 upvotes_instructions: |
```
Comment (Josh-Akers) [inline cves/CVE-2016-5130.yml]: Upvotes: 4

File: cves/CVE-2016-5130.yml (hunk 1 above)
Comment (BenjaminHunt) [reply cves/CVE-2016-5130.yml]: That might help, I'll look into it. Thanks!

File: cves/CVE-2016-5130.yml (hunk 7 above)
Comment (BenjaminHunt) [reply cves/CVE-2016-5130.yml]: I will investigate this further.

File: cves/CVE-2016-1679.yml (hunk 6 above)
Comment (BenjaminHunt) [reply cves/CVE-2016-1679.yml]: It appears that I was looking in the wrong section. That's what it said under the Technical Details, the vulnerability type.

Under the Evaluator Description, it said CWE-416: Use After Free. I will correct this.

File: cves/CVE-2016-5130.yml (hunk 7 above)
Comment (BenjaminHunt) [reply cves/CVE-2016-5130.yml]: I believe I read the wrong bug report. Good find.

**Pull request merged**